device/coordinate_manager.cpp

```diff
@@ -405,6 +405,26 @@ void CoordinateManager::assert_create_coordinate_manager(
     }
 }
 
+void CoordinateManager::shuffle_tensix_harvesting_mask(const std::vector<uint32_t>& harvesting_locations) {
+    std::vector<uint32_t> sorted_harvesting_locations = harvesting_locations;
+    std::sort(sorted_harvesting_locations.begin(), sorted_harvesting_locations.end());
+    size_t new_harvesting_mask = 0;
+    uint32_t pos = 0;
+    while (tensix_harvesting_mask > 0) {
+        if (tensix_harvesting_mask & 1) {
+            uint32_t sorted_position =
+                std::find(
+                    sorted_harvesting_locations.begin(), sorted_harvesting_locations.end(), harvesting_locations[pos]) -
+                sorted_harvesting_locations.begin();
+            new_harvesting_mask |= (1 << sorted_position);
```

Comment:
So tensix_harvesting_mask won't be the original harvesting_mask passed.
Can you comment on the field tensix_harvesting_mask in header file about it not being passed harvesting mask, but rather "normalized", or "noc" harvesting mask, whereas passed one is expected to be a a physical one? Maybe add that comment to the harvesting parameter of the constructor.

Reply:
The resulting harvesting mask is in logical_coordinates right? 
for blackhole for example, each successive bit would correspond to logical coord 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13
which translates to noc0 real coordinate: 1, 2, 3, 4, 5, 6, 7, 10, 11, 12, 13, 14, 15, 16

Reply:
yes like that, I will add a comment